lib/inspector/inspector_service.dart

```diff
@@ -0,0 +1,844 @@
+    if (disposed) {
+      return null;
+    }
+
+    return supplier();
+  }
+
+  void skipIfDisposed(void runnable()) {
+    if (disposed) {
+      return;
+    }
+
+    runnable();
+  }
+
+  Future<SourcePosition> getPropertyLocation(
+      InstanceRef instanceRef, String name) async {
+    final Instance instance = await getInstance(instanceRef);
+    if (instance == null || disposed) {
+      return null;
+    }
+    return getPropertyLocationHelper(instance.classRef, name);
+  }
+
+  Future<SourcePosition> getPropertyLocationHelper(
+      ClassRef classRef, String name) async {
+    final clazz = await inspectorLibrary.getClass(classRef, this);
+    for (FuncRef f in clazz.functions) {
+      // TODO(pq): check for properties that match name.
+      if (f.name == name) {
+        final func = await inspectorLibrary.getFunc(f, this);
+        final SourceLocation location = func.location;
+        throw UnimplementedError(
+            'getSourcePosition not implemented. $location');
+//        return inspectorLibrary.getSourcePosition(
+//            debugProcess, location.script, location.tokenPos, this);
+      }
+    }
+    final ClassRef superClass = clazz.superClass;
+    return superClass == null
+        ? null
+        : getPropertyLocationHelper(superClass, name);
+  }
+
+  Future<DiagnosticsNode> getRoot(FlutterTreeType type) {
+    // There is no excuse to call this method on a disposed group.
+    assert(!disposed);
+    switch (type) {
+      case FlutterTreeType.widget:
+        return getRootWidget();
+      case FlutterTreeType.renderObject:
+        return getRootRenderObject();
+    }
+    throw new Exception('Unexpected FlutterTreeType');
+  }
+
+  /// Invokes a static method on the WidgetInspectorService class passing in the specified
+  /// arguments.
+  ///
+  /// Intent is we could refactor how the API is invoked by only changing this call.
+  Future<InstanceRef> invokeServiceMethodObservatory(String methodName) {
+    return invokeServiceMethodObservatory1(methodName, groupName);
+  }
+
+  Future<InstanceRef> invokeServiceMethodObservatory1(
+      String methodName, String arg1) {
+    return inspectorLibrary.eval(
+      "WidgetInspectorService.instance.$methodName('$arg1')",
+      isAlive: this,
+    );
+  }
+
+  Future<Object> invokeServiceMethodDaemon(String methodName,
+      [String objectGroup]) {
+    return invokeServiceMethodDaemonParams(
+      methodName,
+      {'objectGroup': objectGroup ?? groupName},
+    );
+  }
+
+  Future<Object> invokeServiceMethodDaemonArg(
+      String methodName, String arg, String objectGroup) {
+    return invokeServiceMethodDaemonParams(
+      methodName,
+      {'arg': arg, 'objectGroup': objectGroup},
+    );
+  }
+
+  Future<Object> _callServiceExtension(
+      String extension, Map<String, Object> args) {
+    return inspectorLibrary.addRequest(this, () async {
+      final r = await inspectorService.vmService.callServiceExtension(
```

Comment:
In an ideal world, should all service extension related calls go through the serviceManager? In IntelliJ, does everything go through VmServiceManager (equivalent of DevTools serviceManager)? Curious.

Reply:
all toggle-able service extensions should go through VMServiceManager. These ones in InspectorService are a bit different as they implement a full featured API instead of toggling simple settings on and off.